clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -149,6 +149,82 @@ SYCLInstallationDetector::SYCLInstallationDetector(
     const llvm::opt::ArgList &Args)
     : D(D) {}
 
+static llvm::SmallString<64>
+getLibSpirvBasename(const llvm::Triple &DeviceTriple,
+                    const llvm::Triple &HostTriple) {
+  // Select remangled libclc variant.
+  // Decide long size based on host triple, because offloading targets are going
+  // to match that.
+  // All known windows environments except Cygwin use 32-bit long.
+  llvm::SmallString<64> Result(HostTriple.isOSWindows() &&
+                                       !HostTriple.isWindowsCygwinEnvironment()
+                                   ? "remangled-l32-signed_char.libspirv-"
+                                   : "remangled-l64-signed_char.libspirv-");
```

Comment:
This looks like a vast improvement already, Native CPU will benefit from this as well (we have in-progress patches that would duplicate this that now become just another call to `addLibspirvLinkArgs`). If you're making changes to make this more generally useful, would it be possible to also pick up the correct `signed_char` vs. `unsigned_char` here, or would you prefer to leave that for a later PR?